README.md

```diff
@@ -1 +1,139 @@
+* `JsonOverlay` framework is fairly stable and working well.
+
+* OpenAPI 3.0 type specification, and the generated code passes some
+  early tests.
+
+* Both read and write APIs are fully implemented, though not well
+  tested. No builders have been created as yet.
+
+* Model-level validation is mostly complete, but requires development
+  of extensive tests, as well as a careful inventory of all
+  requirements appearing in the specification (noting all uses of
+  MUST, SHOULD, etc.) to ensure that validation is complete.
+
+* Serialization has not yet been implemented.
+
+* DI for implementation-level validators exists, but validators need
+  to be written. This is likely to be limited to a small number of
+  JSON-level validators, namely detecting where incompatible JSON
+  types appear, and detecting unexpected property names. 
+
+* A handful of high-level tests have been implemented:
+  - *BigParseTest* parses a large model without validation and checks
+    that every value node in the input is accessible in the expected
+    fashion in the resulting model object.
+  - *ExamplesTest* - Parses and validates all example models currently
+    in the
+    [`dm/examples`](https://github.com/OAI/OpenAPI-Specification/tree/dm/examples)
+    branch of the `OAI/OpenAPI-Specification` GitHub repo.
+
+    Many more tests are required!
+
+* Few JavaDocs exist at present, unfortunately. There's an open issue
+  to correct that.
+
+### Packages
+
+_Some of these packages are likely to be refactored out into separate
+component Maven projects._
+
+All packages are currently prefixed by `com.reprezen.`
+
+* `swaggerparser`: Top-level package, directly includes
+  `SwaggerParser` class and some things related to code generation
+  that will probably move elsewehere.
+  
+* `swaggerparser.jsonoverlay`: The JSON Overlay framework underpinning
+  the parser.
+  
+* `swaggerparser.jsonoverlay.std,coll`: Overlay classes for standard
+  scalar types and collection types, respectively.
+  
+* `swaggerparser.jsonoverlay.gen`: Code generators for interfaces and
+  overlay-based implementations of object types
+
+* `swaggerparser.model3`: Generated model interfaces
+
+* `swaggerparser.ovl3`: Generated model implementation classes (using
+  JSON Overlay)
+
+* `swaggerparser.val`: Base classes for validators, and primitive type
+  validators.
+
+* `swaggerparser.val3`: Model-level validators for all OpenAPI
+  objects.
+
+* `swaggerparser.val3.ovl`: Implementation-level validators for
+  OpenAPI objects.
+
+* `swaggerparser.test`: The handful of tests that have been
+  implemented so far. More needed
+
+
+## Contributing
+
+We welcome serious contributors. However, we are at an early and
+fast-changing phase in this project, so coordination with
+collaborators is paricularly important.
+
+If you would like to work with us, please drop us a line at
+info@reprezen.com and let us know where you would like to focus,
+before forking the repo and picking off an open issue.
```

Comment:
I'll change this to suggest that it's for their benefit (so they don't start a line of work that becomes irrelevant or incompatible with other ongoing work, for example). And invite them to open issues, as you suggested.